actk/steps/make_diagnostic_sheet/make_diagnostic_sheet.py

```diff
@@ -0,0 +1,294 @@
+                            dataset["CellImage2DAllProjectionsPath"][index]
+                        )
+                        axes.imshow(img)
+                        axes.set_aspect(1)
+                        index += 1
+                        index2 += 1
+                    else:
+                        axes.axis("off")
+
+            log.info(
+                f"Completed diagnostic sheet for : {metadata} {figure_numbers[i]} "
+            )
+
+            # Savefig
+            fig.savefig(figure_paths[i])
+
+    @staticmethod
+    def _collect_group(
+        row_index: int,
+        row: pd.Series,
+        diagnostic_sheet_dir: Path,
+        overwrite: bool,
+        metadata: str,
+    ) -> Union[DiagnosticSheetResult, DiagnosticSheetError]:
+        # Don't use dask for image reading
+        aicsimageio.use_dask(False)
+
+        try:
+            # Get the ultimate end save paths for grouped plot
+            if row["CellImage2DAllProjectionsPath"]:
+                if row[str(metadata)]:
+                    diagnostic_sheet_save_path = (
+                        diagnostic_sheet_dir / f"{metadata}_{row[metadata]}.png"
+                    )
+                    log.info(f"Generating diagnostic sheet for cell ID: {row.CellId}")
+                else:
+                    # else no path to save
+                    diagnostic_sheet_save_path = None
+
+            # Check skip
+            if not overwrite and diagnostic_sheet_save_path.is_file():
+                log.info(f"Skipping diagnostic sheet for Cell Id: {row.CellId}")
+                return DiagnosticSheetResult(row.CellId, None)
+
+            # Return ready to save image
+            return DiagnosticSheetResult(row.CellId, diagnostic_sheet_save_path)
+        # Catch and return error
+        except Exception as e:
+            log.info(
+                f"Failed to retrieve the CellImage2DAllProjectionsPath"
+                f"for cell ID: {row.CellId}"
+                f"Error: {e}"
+            )
+            return DiagnosticSheetError(row.CellId, str(e))
+
+    @log_run_params
+    def run(
+        self,
+        dataset: Union[str, Path, pd.DataFrame, dd.DataFrame],
+        distributed_executor_address: Optional[str] = None,
+        overwrite: bool = False,
+        metadata: Optional[str] = None,
+        feature: Optional[str] = None,
+        **kwargs,
+    ):
+        """
+        Run a pure function.
+
+        Protected Parameters
+        --------------------
+        distributed_executor_address: Optional[str]
+            An optional executor address to pass to some computation engine.
+        clean: bool
+            Should the local staging directory be cleaned prior to this run.
+            Default: False (Do not clean)
+        debug: bool
+            A debug flag for the developer to use to manipulate how much data runs,
+            how it is processed, etc.
+            Default: False (Do not debug)
```

Comment:
Can these be added to the the `Parameters` section. My bad on splitting them up.